acceptance-tests/src/test/java/tech/pegasys/pantheon/tests/acceptance/dsl/transaction/CallSmartContractFunctionWithRevert.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.tests.acceptance.dsl.transaction;
+
+import tech.pegasys.pantheon.tests.acceptance.dsl.account.Accounts;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.Collections;
+
+import org.web3j.abi.FunctionEncoder;
+import org.web3j.abi.datatypes.Function;
+import org.web3j.crypto.Credentials;
+import org.web3j.protocol.core.methods.response.EthSendTransaction;
+import org.web3j.tx.RawTransactionManager;
+
+public class CallSmartContractFunctionWithRevert implements Transaction<EthSendTransaction> {
```

Comment:
What in this class links to `WithRevert`? ...as it looks like a generic smart contract function invocation

Reply:
renamed to CallSmartContractFunction